### ISSUE TITLE
Rule Change: Add option to suppress `no-underscore-dangle` for private fields

### ISSUE THREAD

**Issue by hfhchan** [eslint/eslint#15720]
### What rule do you want to change?

no-underscore-dangle

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
class FooData {
  private _cachedMap = new Map<string, Foo>();

  constructor(data: Foo[]) {
     data.forEach(row => this._cachedMap.set(row.id, row));
  }

  get(id: string) {
     return this._cachedMap.get(id);
  }
}
```

### What does the rule currently do for this code?

Unexpected dangling '_' in '_cachedMap'.

### What will the rule do after it's changed?

No warning

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by hfhchan** [comment]
Sorry this came in 6 months late.  I believe this is an adjustment to the implementation of #14857.

Underscore properties are discouraged because JavaScript did not have truly private members in the past, but now they do as Class fields, and having an underscore will aid in code clarity in showing that this is indeed a private field.

**Comment by mdjermanovic** [comment]
Hi @hfhchan, thanks for the issue!

> ```js
> private _cachedMap = new Map<string, Foo>();
> ```

This is a TypeScript-specific syntax. Core rules do not provide options related to non-JS syntax, so this is not a feature we could accept.

Since you're using TypeScript, you can check the [`@typescript-eslint/naming-convention`](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md) rule. For TypeScript code, this rule should be preferable over the core naming convention rules.
